packages/state-transition/src/cache/epochCache.ts

```diff
@@ -918,12 +918,22 @@ export class EpochCache {
    * balances are sampled to adjust the probability of the next selection (32 per epoch on average). So to invalidate
    * the prediction the effective of one of those 32 samples should change and change the random_byte inequality.
    */
-  getBeaconProposersNextEpoch(): ValidatorIndex[] {
+  async getBeaconProposersNextEpoch(): Promise<ValidatorIndex[]> {
```

Comment:
using `async` here seems the best way to go, just need to drop some comments saying that this function is technically not part of the state-transition functions, it's just an util to query next epoch's proposers